pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -205,10 +211,15 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		gc.workerLock.Lock()
 		defer gc.workerLock.Unlock()
 
-		// Once we get here, we should not unpause workers until we've successfully synced
+		// Note: The syncing may fail if the CRDs have broken conversion webhooks,
+		// but it should not block the gc working. We retry MaxSyncAttempt to sync
+		// the informers.
 		attempt := 0
-		wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+		err := wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
 			attempt++
+			if attempt > MaxSyncAttempt {
```

Comment:
The outer `wait.Until` already retries

The only reason for the faster inner retry was because the worker loop remained locked until resync completed successfully. Since this is dropping that invariant, I don't see a reason to retry the inner loop a *few* times on resync timeout/failure. It seems simpler to drop the inner `wait.PollImmediateUntil` and fallback to the outer `wait.Until` loop on resync timeout/failure

Reply:
Yes, you're right! The ouiter `wati.Util` is enough.